src/components/BackToTopButton/index.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { Component } from 'react';
+import { BackToTopStyles } from './index.styles';
+
+export default class BackToTopButton extends Component {
+  backToTopButton: any;
+  constructor(props: any) {
+    super(props);
+    this.backToTopButton = React.createRef();
+  }
+
+  componentDidMount() {
+    document.addEventListener('scroll', () => {
+      if (window.scrollY > 1000) {
+        this.backToTopButton.current.classList.add('open');
```

Comment:
This should be state instead of using a ref

Reply:
like this?
```
export default class BackToTopButton extends Component<any> {

  state = {
    open: false,
  };

  componentDidMount() {
    document.addEventListener('scroll', () => {
      if (window.scrollY > 1000) {
        this.setState({
          open: true,
        });
      } else {
        this.setState({
          open: false,
        });
      }
    });
  }

  render() {
    return (
      this.state.open &&
      <BackToTopStyles
        id="back-to-top-button"
        title="Back to top"
        onClick={() => window.scrollTo(0, 0)}
        className="open"
      >
        <span>&nbsp;</span>
        <span>&nbsp;</span>
      </BackToTopStyles>
    );
  }
}
```
If I do it like this I lose the nice 'slide in' effect.

Reply:
😩 Shouldn't code when tired..
```
  state = {
    open: '',
  };

  componentDidMount() {
    document.addEventListener('scroll', () => {
      if (window.scrollY > 1000) {
        this.setState({
          open: 'open',
        });
      } else {
        this.setState({
          open: '',
        });
      }
    });
  }

  render() {
    return (
      <BackToTopStyles
        id="back-to-top-button"
        title="Back to top"
        onClick={() => window.scrollTo(0, 0)}
        className={this.state.open}
      >
        <span>&nbsp;</span>
        <span>&nbsp;</span>
      </BackToTopStyles>
    );
  }
}
```
Maybe this is ok, I'll push this.